src/librustc_mir/borrow_check/nll/invalidation.rs

```diff
@@ -0,0 +1,955 @@
+            }
+            TerminatorKind::DropAndReplace {
+                location: ref drop_place,
+                value: ref new_value,
+                target: _,
+                unwind: _,
+            } => {
+                self.mutate_place(
+                    ContextKind::DropAndReplace.new(location),
+                    drop_place,
+                    Deep,
+                    JustWrite,
+                );
+                self.consume_operand(
+                    ContextKind::DropAndReplace.new(location),
+                    new_value,
+                );
+            }
+            TerminatorKind::Call {
+                ref func,
+                ref args,
+                ref destination,
+                cleanup: _,
+            } => {
+                self.consume_operand(ContextKind::CallOperator.new(location), func);
+                for arg in args {
+                    self.consume_operand(ContextKind::CallOperand.new(location), arg);
+                }
+                if let Some((ref dest, _ /*bb*/)) = *destination {
+                    self.mutate_place(
+                        ContextKind::CallDest.new(location),
+                        dest,
+                        Deep,
+                        JustWrite,
+                    );
+                }
+            }
+            TerminatorKind::Assert {
+                ref cond,
+                expected: _,
+                ref msg,
+                target: _,
+                cleanup: _,
+            } => {
+                self.consume_operand(ContextKind::Assert.new(location), cond);
+                use rustc::mir::interpret::EvalErrorKind::BoundsCheck;
+                if let BoundsCheck { ref len, ref index } = *msg {
+                    self.consume_operand(ContextKind::Assert.new(location), len);
+                    self.consume_operand(ContextKind::Assert.new(location), index);
+                }
+            }
+            TerminatorKind::Yield {
+                ref value,
+                resume: _,
+                drop: _,
+            } => {
+                self.consume_operand(ContextKind::Yield.new(location), value);
+
+                // ** FIXME(bob_twinkles) figure out what the equivalent of this is
+                // if self.movable_generator {
+                //     // Look for any active borrows to locals
+                //     let borrow_set = self.borrow_set.clone();
+                //     flow_state.with_outgoing_borrows(|borrows| {
+                //         for i in borrows {
+                //             let borrow = &borrow_set[i];
+                //             self.check_for_local_borrow(borrow, span);
+                //         }
+                //     });
+                // }
+            }
+            TerminatorKind::Resume | TerminatorKind::Return | TerminatorKind::GeneratorDrop => {
+                // ** FIXME(bob_twinkles) figure out what the equivalent of this is
```

Comment:
We want to invalidate all loans that refer to content owned by the current stack frame; I'm actually not sure why this uses `with_outgoing_borrows`, I think for our purposes the current location is fine (`self.location_table.start_index(loc)`).

That is what I did in my branch that integrates with dataflow, anyway:

https://github.com/nikomatsakis/rust/blob/5656960ea0b62624341a10c175ab9d154a5fe94c/src/librustc_mir/borrow_check/mod.rs#L606-L618